### PR TITLE
[backend] Improve RSS feed converter to support AtomV1 and more

### LINKED ISSUE #4339: Improve RSS feed converter to support AtomV1 and complex categories/description
## Use case

Improve RSS feed converter to support complex categories and description.
Example of feeds that fail that need to be supported.
https://support.citrix.com/feed/products/all/securitybulletins.rss
https://www.oracle.com/ocom/groups/public/@otn/documents/webcontent/rss-otn-sec.xml

Add support for atomV1 feed like https://cloud.google.com/feeds/google-cloud-security-bulletins.xml

### PR DESCRIPTION
**Pull request by richard-julien**
See https://github.com/OpenCTI-Platform/opencti/issues/4339

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/manager/ingestionManager.ts [hunk 1]
```diff
@@ -50,50 +49,90 @@ const asArray = (data: unknown) => {
...
+export const rssDataParser = async (turndownService: TurndownService, data: convertableToString, current_state_date: Date | undefined): Promise<DataItem[]> => {
+  const xmlData = await xmlParse(data, { explicitArray: false, trim: true, explicitCharkey: true, mergeAttrs: true });
+  if (xmlData?.feed) { // Atom V1
+    const entries = asArray(xmlData.feed.entry);
+    return entries.map((entry) => rssItemV1Convert(turndownService, xmlData.feed, entry))
+      .filter((e) => isNotEmptyField(e.title)) // Title is mandatory
+      .filter((e) => isNotEmptyField(e.pubDate) && e.pubDate.isValid()) // A valid date is also mandatory
+      .filter((e) => isEmptyField(current_state_date) || e.pubDate.isAfter(current_state_date))
+      .sort((a, b) => a.pubDate.diff(b.pubDate));
+  }
```
Comment (Goumies) [inline opencti-platform/opencti-graphql/src/manager/ingestionManager.ts]: We could extract this logic in a function and name it to explicit the comment on the condition

File: opencti-platform/opencti-graphql/src/manager/ingestionManager.ts [hunk 2]
```diff
@@ -50,50 +49,90 @@ const asArray = (data: unknown) => {
...
+  if (xmlData?.rss) { // Atom V2
+    const channels = asArray(xmlData?.rss?.channel);
+    return channels
+      .map((channel) => asArray(channel.item).map((item) => rssItemV2Convert(turndownService, channel, item))).flat()
+      .filter((e) => isNotEmptyField(e.title)) // Title is mandatory
+      .filter((e) => isNotEmptyField(e.pubDate) && e.pubDate.isValid()) // A valid date is also mandatory
+      .filter((e) => isEmptyField(current_state_date) || e.pubDate.isAfter(current_state_date))
+      .sort((a, b) => a.pubDate.diff(b.pubDate));
+  }
```
Comment (Goumies) [inline opencti-platform/opencti-graphql/src/manager/ingestionManager.ts]: Same thing here

File: opencti-platform/opencti-graphql/tests/01-unit/database/rss-test.js [hunk 3]
```diff
@@ -0,0 +1,324 @@
+import { describe, expect, it } from 'vitest';
+import TurndownService from 'turndown';
+import { rssDataParser } from '../../../src/manager/ingestionManager';
+
+// region https://www.oracle.com/ocom/groups/public/@otn/documents/webcontent/rss-otn-sec.xml
```
Comment (Goumies) [inline opencti-platform/opencti-graphql/tests/01-unit/database/rss-test.js]: Maybe the initialization of those variables could be extracted to a single file or file by variable.

Comment (richard-julien) [reply opencti-platform/opencti-graphql/tests/01-unit/database/rss-test.js]: Extracted to xml files in data

File: opencti-platform/opencti-graphql/src/manager/ingestionManager.ts (hunk 1 above)
Comment (richard-julien) [reply opencti-platform/opencti-graphql/src/manager/ingestionManager.ts]: Filtering extracted to specific method.

**Review by Goumies — APPROVED** [review]

**Pull request merged**